### PR TITLE
:sparkles: implements file repo field displaynames

### PR DESCRIPTION
**Pull request by hlminh2000**
**Description of changes**

pulls fields' display names from the gateway for:
- facet panel
- table

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Add copyrights to new files
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 1]
```diff
@@ -143,7 +146,15 @@ const useFileFacetQuery = (
...
-  const [expandedFacets, setExpandedFacets] = React.useState(concat(presetFacets, fileIDSearch));
+  const {
+    data: fieldDisplayNames,
+    loading: loadingFieldDisplayNames,
+  } = useFileCentricFieldDisplayName();
+  const presetFacets = createPresetFacets(fieldDisplayNames);
+  const [expandedFacets, setExpandedFacets] = React.useState(
+    [...presetFacets, fileIDSearch].map(facet => facet.facetPath),
+  );
+  console.log('presetFacets: ', presetFacets);
```
Comment (anncatton) [inline components/pages/file-repository/FacetPanel/index.tsx:157]: Guessing we can remove this log? :)

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 2]
```diff
@@ -272,74 +284,79 @@ export default () => {
...
-              clickHandler(type);
-            },
-            isExpanded: expandedFacets.includes(type),
-            subMenuName: capitalize(type.name),
-          };
+        {!loadingFieldDisplayNames &&
+          presetFacets.map(type => {
+            const commonFacetProps = {
+              onClick: e => {
+                console.log('e: ', e);
```
Comment (anncatton) [inline components/pages/file-repository/FacetPanel/index.tsx:291]: same as above :)

File: components/pages/file-repository/hooks/useFileCentricFieldDisplayName.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const useFileCentricFieldDisplayName = (): {
+  data: { [k in FileCentricDocumentField]: string };
+  loading: boolean;
+} => {
+  const { data, loading } = useQuery<{ file: { extended: ExtendedMapping } }>(
+    FILE_CENTRIC_EXTENDED_MAPPING,
+  );
+
+  return {
+    // @ts-ignore this will result in unknown fields being present in run time, but that's why we have you Typescript!
```
Comment (anncatton) [inline components/pages/file-repository/hooks/useFileCentricFieldDisplayName.tsx:19]: 😀

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 4]
```diff
@@ -54,52 +53,56 @@ import {
...
   justify-content: space-between;
 `;
 
-const presetFacets: Array<FacetDetails> = [
+const createPresetFacets = (
+  displayNames: ReturnType<typeof useFileCentricFieldDisplayName>['data'],
+): Array<FacetDetails> => [
   {
-    name: 'program id',
+    name: displayNames['study_id'] || '__',
```
Comment (anncatton) [inline components/pages/file-repository/FacetPanel/index.tsx:69]: just curious why `__` is the alternative?

File: components/pages/file-repository/hooks/useFileCentricFieldDisplayName.tsx (hunk 3 above)
Comment (hlminh2000) [reply components/pages/file-repository/hooks/useFileCentricFieldDisplayName.tsx:19]: lol yeah I like to leave comments explaining why when I use `@ts-ignore` :P makes every exception come with an explanation

File: components/pages/file-repository/FacetPanel/index.tsx (hunk 4 above)
Comment (hlminh2000) [reply components/pages/file-repository/FacetPanel/index.tsx:69]: yeah I could have done better here, will update with something that makes more sense :)

File: components/pages/file-repository/FileTable/TsvDownloadButton.tsx [hunk 5]
```diff
@@ -114,7 +114,7 @@ export default ({
         : defaultFilters,
     ],
   };
-  const onItemClick: DownloadButtonProps<DownloadOptionValues>['onItemClick'] = item => {
+  const onItemClick: DownloadButtonProps<DownloadOptionValues>['onItemClick'] = (item) => {
```
Comment (anncatton) [inline components/pages/file-repository/FileTable/TsvDownloadButton.tsx:117]: is this new formatting from the latest prettier update you made?

File: components/pages/file-repository/types.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export enum FileCentricDocumentField {
```
Comment (anncatton) [inline components/pages/file-repository/types.ts:1]: Do you think we'll add more types in this file? Or should we just call it `enums.ts` :)

Comment (hlminh2000) [reply components/pages/file-repository/types.ts:1]: I think `types.ts` is fine, the fact that this one is an enum is a small implementation detail

**Pull request merged**